### PR TITLE
Add token expiry metrics

### PR DESCRIPTION
**Pull request by t4niwa**
# Description

Add token_expires_in_seconds metrics

## example 1

```bash
$ curl -sfS localhost:9793/metrics | grep 'token_expires_in_seconds'
# HELP token_expires_in_seconds Indicates remaining time until the token expires.
# TYPE token_expires_in_seconds gauge
token_expires_in_seconds{domain="domain1",role="role1",type="roletoken"} 14120.64295
token_expires_in_seconds{domain="domain1",role="role1",type="accesstoken"} 12754.642959
token_expires_in_seconds{domain="domain2",role="role2",type="roletoken"} 12754.642952
```

## example 2

```bash
$ curl -sfS localhost:9793/metrics | grep 'token_expires_in_seconds'
# HELP token_expires_in_seconds Indicates remaining time until the token expires.
# TYPE token_expires_in_seconds gauge
token_expires_in_seconds{domain="domain",role="users",type="roletoken"} 13877.364748
token_expires_in_seconds{domain="domain",max_expiry="10000",role="role",type="roletoken"} 9510.364777
token_expires_in_seconds{domain="domain",min_expiry="10000",role="role",type="roletoken"} 13923.364768
token_expires_in_seconds{domain="domain",max_expiry="100000",role="role",type="roletoken"} 99486.364787
token_expires_in_seconds{domain="domain",max_expiry="10000",min_expiry="10000",role="role",type="roletoken"} 9560.364757
```

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Passed all pipeline checking

## Checklist for maintainer

- Use `Squash and merge`
- Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- Delete the branch after merge

### CODE REVIEW

File: pkg/token/cache.go [hunk 1]
```diff
@@ -132,3 +140,50 @@ func (c *LockedTokenCache) Clear() {
 	}
 	c.memoryUsage = 0
 }
+
+var (
+	tokenExpiresInMetric = "token_expires_in_seconds"
+	tokenExpiresInHelp   = "Indicates remaining time until the token expires."
```
Comment (WindzCUHK) [inline pkg/token/cache.go:146]:
```suggestion
	tokenExpiresInHelp   = "Indicates remaining time until the token expires"
```

File: pkg/token/cache.go [hunk 2]
```diff
@@ -132,3 +140,50 @@ func (c *LockedTokenCache) Clear() {
 	}
```
Comment (WindzCUHK) [inline pkg/token/cache.go:140]:
```suggestion
		// TODO: reset metrics on delete
	}
```
add comment

File: pkg/token/cache.go [hunk 3]
```diff
@@ -132,3 +140,50 @@ func (c *LockedTokenCache) Clear() {
...
+	ch <- prometheus.NewDesc(tokenExpiresInMetric, tokenExpiresInHelp, labelKeys, prometheus.Labels{
+		"type": c.tokenType,
+	})
+}
+
+func (c *LockedTokenCache) Collect(ch chan<- prometheus.Metric) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	for k, t := range c.cache {
```
Comment (WindzCUHK) [inline pkg/token/cache.go:160]:
```suggestion
	for k, t := range c.cache {
	
	// skip placeholder token added during daemon creation
	// if t.domain = "" { continue }
```
https://github.com/AthenZ/k8s-athenz-sia/blob/6ac184b46db85a5e5fed275ddc3976b1320bc227/pkg/token/daemon.go#L73-L78

Comment (t4niwa) [reply pkg/token/cache.go:162]: @WindzCUHK fixed in https://github.com/AthenZ/k8s-athenz-sia/pull/77/commits/732ace5f79e2f591ea5d7a96bd43c702a934453c

File: pkg/token/cache.go (hunk 1 above)
Comment (t4niwa) [reply pkg/token/cache.go]: @WindzCUHK fixed in https://github.com/AthenZ/k8s-athenz-sia/pull/77/commits/732ace5f79e2f591ea5d7a96bd43c702a934453c

**Review by WindzCUHK — APPROVED** [review]

**Pull request merged**